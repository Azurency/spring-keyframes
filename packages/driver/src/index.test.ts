import { default as spring } from './index'

it('returns an array of keyframe strings, duration, and ease', () => {
  const [animation, duration, ease] = spring(
    { width: 100, x: 0, y: 100, scale: 0 },
    { width: 200, x: 20, y: 50, scale: 1 }
  )

  expect(animation[0]).toMatchSnapshot()
  expect(duration).toBe('750ms')
  expect(ease).toMatchSnapshot()
})

it('returns an array of keyframe strings, and duration for a long spring', () => {
  const [animation, duration] = spring(
    { width: 100 },
    { width: 200 },
    { stiffness: 400, damping: 3 }
  )

  expect(animation[0]).toMatchSnapshot()
  expect(duration).toBe('3250ms')
})

it('returns an array of keyframe strings, and duration for a short spring', () => {
  const [animation, duration] = spring(
    { width: 0 },
    { width: 200 },
    { stiffness: 2, damping: 3 }
  )

  expect(animation[0]).toMatchSnapshot()
  expect(duration).toBe('5350ms')
})

it('returns an array of keyframe strings, and duration for a tweened animation', () => {
  const [animation, duration] = spring(
    { opacity: 0 },
    { opacity: 1 },
    { stiffness: 100, damping: 3 }
  )

  expect(animation[0]).toMatchSnapshot()
  expect(duration).toBe('3133.33ms')
})

it('returns an array of keyframe strings, and duration for a tweened and sprung animation', () => {
  const [animations, duration] = spring(
    { opacity: 0, x: 0 },
    { opacity: 1, x: 100 },
    { stiffness: 100, damping: 3 }
  )

  expect(animations[0]).toMatchSnapshot()
  expect(animations[1]).toMatchSnapshot()
  expect(duration).toBe('3133.33ms')
})

it('returns an array of valid keyframes for camelCase properties', () => {
  const [animations, duration] = spring(
    { borderRadius: 0 },
    { borderRadius: 20 },
    { stiffness: 100, damping: 3 }
  )

  expect(animations[0]).toMatchSnapshot()
  expect(duration).toBe('3133.33ms')
})

it('returns tweened animations for all tweened property values', () => {
  const [animations, duration] = spring(
    { opacity: 0, x: 0 },
    { opacity: 1, x: 100 },
    { stiffness: 100, damping: 3, tweenedProps: ['opacity', 'x'] }
  )

  expect(animations[0]).toMatchSnapshot()
  expect(duration).toBe('3133.33ms')
})

it('returns tweened animations for custom tweened property values', () => {
  const [animations, duration] = spring(
    { opacity: 0, x: 0 },
    { opacity: 1, x: 100 },
    { stiffness: 100, damping: 3, tweenedProps: ['x'] }
  )

  expect(animations[0]).toMatchSnapshot()
  expect(animations[1]).toMatchSnapshot()
  expect(duration).toBe('3133.33ms')
})

it('returns an array of valid keyframes for backgroundColor', () => {
  const [animations, duration] = spring(
    { backgroundColor: 'red' },
    { backgroundColor: 'blue' },
    { stiffness: 100, damping: 3 }
  )

  expect(animations[0]).toMatchSnapshot()
  expect(duration).toMatchSnapshot()
})

it('returns a velocity for a in-progress animation', () => {
  const [animations, duration, ease, convert] = spring(
    { x: 0 },
    { x: 400 },
    { stiffness: 100, damping: 2 }
  )

  expect(convert(400)).toBe(-0.10951148823495502)
  expect(ease).toBeDefined()
  expect(animations).toBeDefined()
  expect(duration).toBeDefined()
})
