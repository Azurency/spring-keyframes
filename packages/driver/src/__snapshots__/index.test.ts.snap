// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns an array of keyframe strings, and duration for a long spring 1`] = `
"0% {width: 100px;}
 4.62% {width: 279.74px;}
 9.23% {width: 137.01px;}
 13.85% {width: 249.28px;}
 18.97% {width: 161.15px;}
 23.59% {width: 230.83px;}
 28.21% {width: 175.77px;}
 33.33% {width: 218.89px;}
 37.95% {width: 184.94px;}
 42.56% {width: 211.89px;}
 47.18% {width: 190.7px;}
 52.31% {width: 207.34px;}
 56.92% {width: 194.18px;}
 61.54% {width: 204.57px;}
 66.67% {width: 196.43px;}
 71.28% {width: 202.84px;}
 75.9% {width: 197.76px;}
 80.51% {width: 201.75px;}
 85.64% {width: 198.61px;}
 90.26% {width: 201.1px;}
 94.87% {width: 199.14px;}
 100% {width: 200px;}"
`;

exports[`returns an array of keyframe strings, and duration for a short spring 1`] = `
"0% {width: 0px;}
 100% {width: 200px;}"
`;

exports[`returns an array of keyframe strings, and duration for a tweened and sprung animation 1`] = `
"0% {transform: translate3d(0px, 0px, 0px);}
 9.57% {transform: translate3d(161.87px, 0px, 0px);}
 19.68% {transform: translate3d(61.76px, 0px, 0px);}
 29.79% {transform: translate3d(123.61px, 0px, 0px);}
 39.89% {transform: translate3d(85.44px, 0px, 0px);}
 50% {transform: translate3d(108.97px, 0px, 0px);}
 59.57% {transform: translate3d(94.45px, 0px, 0px);}
 69.68% {transform: translate3d(103.43px, 0px, 0px);}
 79.79% {transform: translate3d(97.88px, 0px, 0px);}
 89.89% {transform: translate3d(101.31px, 0px, 0px);}
 100% {transform: translate3d(100px, 0px, 0px);}"
`;

exports[`returns an array of keyframe strings, and duration for a tweened and sprung animation 2`] = `
"0% {opacity: 0;}
 100% {opacity: 1;}"
`;

exports[`returns an array of keyframe strings, and duration for a tweened animation 1`] = `
"0% {opacity: 0;}
 100% {opacity: 1;}"
`;

exports[`returns an array of keyframe strings, duration, and ease 1`] = `
"0% {width: 100px; transform: translate3d(0px, 100px, 0px) scale3d(0, 0, 1);}
 33.33% {width: 218.68px; transform: translate3d(23.74px, 40.66px, 0px) scale3d(1.19, 1.19, 1);}
 66.67% {width: 196.51px; transform: translate3d(19.3px, 51.75px, 0px) scale3d(0.97, 0.97, 1);}
 100% {width: 200px; transform: translate3d(20px, 50px, 0px) scale3d(1, 1, 1);}"
`;

exports[`returns an array of valid keyframes for backgroundColor 1`] = `
"0% {background-color: red;}
 100% {background-color: blue;}"
`;

exports[`returns an array of valid keyframes for backgroundColor 2`] = `"3133.33ms"`;

exports[`returns an array of valid keyframes for camelCase properties 1`] = `
"0% {border-radius: 0px;}
 9.57% {border-radius: 32.37px;}
 19.68% {border-radius: 12.35px;}
 29.79% {border-radius: 24.72px;}
 39.89% {border-radius: 17.09px;}
 50% {border-radius: 21.79px;}
 59.57% {border-radius: 18.89px;}
 69.68% {border-radius: 20.69px;}
 79.79% {border-radius: 19.58px;}
 89.89% {border-radius: 20.26px;}
 100% {border-radius: 20px;}"
`;

exports[`returns tweened animations for all tweened property values 1`] = `
"0% {opacity: 0; x: 0px;}
 100% {opacity: 1; x: 100px;}"
`;

exports[`returns tweened animations for custom tweened property values 1`] = `
"0% {opacity: 0;}
 9.57% {opacity: 1.62;}
 19.68% {opacity: 0.62;}
 29.79% {opacity: 1.24;}
 39.89% {opacity: 0.85;}
 50% {opacity: 1.09;}
 59.57% {opacity: 0.94;}
 69.68% {opacity: 1.03;}
 79.79% {opacity: 0.98;}
 89.89% {opacity: 1.01;}
 100% {opacity: 1;}"
`;

exports[`returns tweened animations for custom tweened property values 2`] = `
"0% {x: 0px;}
 100% {x: 100px;}"
`;
